source/extensions/filters/udp/udp_proxy/udp_proxy_filter.cc

```diff
@@ -56,23 +61,14 @@ Network::FilterStatus UdpProxyFilter::onReceiveError(Api::IoError::IoErrorCode)
 }
 
 UdpProxyFilter::ClusterInfo::ClusterInfo(UdpProxyFilter& filter,
-                                         Upstream::ThreadLocalCluster& cluster)
+                                         Upstream::ThreadLocalCluster& cluster,
+                                         SessionStorageType&& sessions)
     : filter_(filter), cluster_(cluster),
-      cluster_stats_(generateStats(cluster.info()->statsScope())),
+      cluster_stats_(generateStats(cluster.info()->statsScope())), sessions_(std::move(sessions)),
       member_update_cb_handle_(cluster.prioritySet().addMemberUpdateCb(
           [this](const Upstream::HostVector&, const Upstream::HostVector& hosts_removed) {
             for (const auto& host : hosts_removed) {
-              // This is similar to removeSession() but slightly different due to removeSession()
-              // also handling deletion of the host to session map entry if there are no sessions
-              // left. It would be nice to unify the logic but that can be cleaned up later.
-              auto host_sessions_it = host_to_sessions_.find(host.get());
-              if (host_sessions_it != host_to_sessions_.end()) {
-                for (const auto& session : host_sessions_it->second) {
-                  ASSERT(sessions_.count(session) == 1);
-                  sessions_.erase(session);
-                }
-                host_to_sessions_.erase(host_sessions_it);
-              }
+              removeHostSessions(host);
```

Comment:
Can this be inlined again since it's only called once? Can you inline/revert other things that are only called once? It will be simpler to review the diff.

Reply:
done